docs/HKT.md

```diff
@@ -0,0 +1,168 @@
+# HKT, URI2HKT, HKTS, HKTAs
+
+## Introduction
+
+Here's the definition of `HKT`
+
+```ts
+// HKT.ts
+
+export interface HKT<URI, A> {
+  readonly _URI: URI
+  readonly _A: A
+}
+```
+
+and `Functor`
+
+```ts
+// Functor.ts
+
+export interface Functor<F> {
+  readonly URI: F
```

Comment:
Does Functor need a URI field?

Reply:
With recent versions of TypeScript? Not sure

Reply:
I got it - this `URI` is used when creating an _instance_ of `Functor`. And `_URI` is used to satisfy `HKT` interface when implementing it with custom type class. I think it should be also in the docs otherwise it's not clear what's the difference between `URI` and `_URI`.